x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -125,66 +143,72 @@ public boolean isStopped() {
     }
 
     public synchronized void start(ActionListener<Response> listener) {
-        final IndexerState prevState = indexer.getState();
-        if (prevState != IndexerState.STOPPED) {
-            // fails if the task is not STOPPED
-            listener.onFailure(new ElasticsearchException("Cannot start task for data frame transform [{}], because state was [{}]",
-                    transform.getId(), prevState));
-            return;
-        }
-
         final IndexerState newState = indexer.start();
-        if (newState != IndexerState.STARTED) {
+        if (Arrays.stream(RUNNING_STATES).noneMatch(newState::equals)) {
             listener.onFailure(new ElasticsearchException("Cannot start task for data frame transform [{}], because state was [{}]",
                     transform.getId(), newState));
             return;
         }
-
-        final DataFrameTransformState state = new DataFrameTransformState(IndexerState.STOPPED, indexer.getPosition(), generation.get());
-
-        logger.debug("Updating state for data frame transform [{}] to [{}][{}]", transform.getId(), state.getIndexerState(),
-                state.getPosition());
-        updatePersistentTaskState(state,
-                ActionListener.wrap(
-                        (task) -> {
-                            auditor.info(transform.getId(), "Updated state to [" + state.getIndexerState() + "]");
-                            logger.debug("Successfully updated state for data frame transform [" + transform.getId() + "] to ["
-                                    + state.getIndexerState() + "][" + state.getPosition() + "]");
-                            listener.onResponse(new StartDataFrameTransformTaskAction.Response(true));
-                        }, (exc) -> {
-                            // We were unable to update the persistent status, so we need to shutdown the indexer too.
-                            indexer.stop();
-                            listener.onFailure(new ElasticsearchException("Error while updating state for data frame transform ["
+        stateReason.set(null);
+        taskState.set(DataFrameTransformTaskState.STARTED);
+        failureCount.set(0);
+
+        final DataFrameTransformState state = new DataFrameTransformState(
+            DataFrameTransformTaskState.STARTED,
+            IndexerState.STOPPED,
+            indexer.getPosition(),
+            generation.get(),
+            null);
+        auditor.info(transform.getId(), "");
```

Comment:
Empty message. I guess this is redundant as there is an audit message line 167 after the state has been set